aten/src/ATen/detail/PrivateUse1HooksInterface.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <c10/core/Device.h>
+#include <c10/util/Exception.h>
+namespace at {
+
+struct TORCH_API PrivateUse1HooksInterface {
+  virtual ~PrivateUse1HooksInterface() = default;
+  virtual const at::Generator& getDefaultGenerator(c10::DeviceIndex device_index) {
+    C10_THROW_ERROR(NotImplementedError, "must be inherited to implement.");
```

Comment:
You can use TORCH_CHECK or TORCH_CHECK_NOT_IMPLEMENTED (if you want a NotImplementedError type on the Python side) here.
Also the error should be that there is no PrivateUse1 device registered (similar to https://github.com/pytorch/pytorch/blob/02cd971e958f4e2f9073e5021501e88cd0460731/aten/src/ATen/detail/CUDAHooksInterface.h#L73)?